k8s/pkg/prefixcollector/collector_server.go

```diff
@@ -0,0 +1,163 @@
+package prefixcollector
+
+import (
+	"fmt"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/prefix_pool"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"net"
+)
+
+type SubnetWatcher struct {
+	subnetCh <-chan *net.IPNet
+	stopCh   chan struct{}
+}
+
+func (s *SubnetWatcher) Stop() {
+	close(s.stopCh)
+}
+
+func (s *SubnetWatcher) ResultChan() <-chan *net.IPNet {
+	return s.subnetCh
+}
+
+type keyFunc func(event watch.Event) (string, error)
+type subnetFunc func(event watch.Event) (*net.IPNet, error)
+
+func WatchPodCIDR(clientset *kubernetes.Clientset) (*SubnetWatcher, error) {
+	nodeWatcher, err := clientset.CoreV1().Nodes().Watch(metav1.ListOptions{})
+	if err != nil {
+		logrus.Error(err)
+		return nil, err
+	}
+
+	nodeCast := func(obj runtime.Object) (*v1.Node, error) {
+		node, ok := obj.(*v1.Node)
+		if !ok {
+			return nil, fmt.Errorf("Casting object to *v1.Service failed. Object: %v", obj)
+		}
+		return node, nil
+	}
+
+	keyFunc := func(event watch.Event) (string, error) {
+		node, err := nodeCast(event.Object)
+		if err != nil {
+			return "", err
+		}
+		return node.Name, nil
+	}
+
+	subnetFunc := func(event watch.Event) (*net.IPNet, error) {
+		node, err := nodeCast(event.Object)
+		if err != nil {
+			return nil, err
+		}
+		_, ipNet, err := net.ParseCIDR(node.Spec.PodCIDR)
+		if err != nil {
+			return nil, err
+		}
+		return ipNet, nil
+	}
+
+	return watchSubnet(nodeWatcher, keyFunc, subnetFunc)
+}
+
+func WatchServiceIpAddr(clientset *kubernetes.Clientset) (*SubnetWatcher, error) {
+	serviceWatcher, err := clientset.CoreV1().Services("default").Watch(metav1.ListOptions{})
```

Comment:
Same as the previous one about using the "default" namespace 👍

Reply:
Thank you! Will fix